### PR TITLE
Show a setup warning in case S3 object storage is used as primary storage

### PR DESCRIPTION
**Pull request by MorrisJobke**
* checks for at least 50 GB of free space

As discussed @rullzer

### CODE REVIEW

**Review by icewind1991 — APPROVED** [review]

**Review by rullzer — APPROVED** [review]

File: core/js/setupchecks.js [hunk 1]
```diff
@@ -450,6 +450,18 @@
 							type: OC.SetupChecks.MESSAGE_TYPE_WARNING
 						})
 					}
+					if (!data.isEnoughTempSpaceAvailableIfS3PrimaryStorageIsUsed) {
+						messages.push({
+							msg: t(
+								'core',
+								'This instance uses an S3 based object store as primary storage. The uploaded files are stored temporarily on the server and thus it is recommended to have 50 GB of free space available in the temp directory of PHP. Check the logs for full details about the path and the available space. To improve this please change the temporary directory in the php.ini or make more space available in that path.',
+								{
+									docLink: oc_defaults.docPlaceholderUrl.replace('PLACEHOLDER', 'admin-mysql-utf8mb4'),
```
Comment (SystemKeeper) [inline core/js/setupchecks.js]: @MorrisJobke  Is the docLink a copy&paste mistake? I don't see a placeholder in the message and linking to 'admin-mysql-utf8mb4' is obviously wrong in this case.

Comment (MorrisJobke) [reply core/js/setupchecks.js]: Thanks for noticing - yes this is an c&p mistake.

Comment (MorrisJobke) [reply core/js/setupchecks.js]: Here is the fix: https://github.com/nextcloud/server/pull/14620

**Comment by MorrisJobke** [comment]
Backport together with #14620 once #14857 is in.

**Pull request merged**